### DISCUSSION TITLE
Use updated libraries, board, IDE and OS (and everything) for widgets.

### DISCUSSION THREAD

**Discussion by ashoksharma842** [Bodmer/TFT_eSPI#2016] [General]
I am trying out the widgets on esp32 devkit and 320x240 TFT LCD touchscreen. Previously I used this library for simple stuff like drawing shapes and text, it was working fine.
So I installed widget library from Arduino IDE.
It gives error "LittleFS.h" not found.
Google and installed.
Still same error.
Somewhere it said it's "LITTLEFS.h" now.
So changed it.
Then comes new error!
After some reading got to know LittleFS is now part or esp32 package 2.0. so update it.
Arduino IDE doesn't give any update after version 1.6.
So update the preference from "stable" to "development" release.
Still not giving update.
So I updated the IDE, did the process again.
Now a new error regarding "py installer".
After searching it says window 7 had this issue update to windows 10!
Then found a window update that can solve the issue.
Now it works fine, for button widgets.
But still facing issues with slider issues.

**Comment by Bodmer** [comment]
The RP2040, ESP32 and ESP8266 board packages now have LittleFS built in so I am surprised there is a problem. You will need ESP32 board package 2.0.0 or later, you will need to update the json web link in the IDE if you are using an older version of the board package. Details here:
https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html#installing-using-arduino-ide


You can raise an issue here is there is a problem:
https://github.com/Bodmer/TFT_eWidget/issues

**Reply by ashoksharma842** [reply]
Yep, I got that.
I mentioned my journey to the solution, just in case anyone else (newbie) faces something like this, he can get some help.

**Reply by Bodmer** [reply]
OK, thanks for the help.
